src/main/java/com/chikli/hudson/plugin/naginator/NaginatorPublisher.java

```diff
@@ -144,4 +167,101 @@ public Notifier newInstance(StaplerRequest req, JSONObject formData) throws Form
 
     private static final Logger LOGGER = Logger.getLogger(NaginatorPublisher.class.getName());
 
+    public static class Action extends NaginatorScheduleAction {
+        private final String regexpForRerun;
+        private final boolean rerunIfUnstable;
+        private final boolean checkRegexp;
+        
+        public Action(NaginatorPublisher publisher) {
+            super(publisher.getMaxSchedule(), publisher.getDelay(), publisher.isRerunMatrixPart());
+            this.regexpForRerun = publisher.getRegexpForRerun();
+            this.rerunIfUnstable = publisher.isRerunIfUnstable();
+            this.checkRegexp = publisher.isCheckRegexp();
+        }
+        
+        @Nullable
```

Comment:
as far as i understand FB, any field is Nullabble by default, if you want enforce null check, then place `CheckForNull`